metaflow/plugins/airflow/plumbing/airflow_xcom_push.py

```diff
@@ -0,0 +1,18 @@
+# How does xcom push work in K8s Operator : https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html#how-does-xcom-work
+import os
+import json
+
+K8S_XCOM_DIR_PATH = "/airflow/xcom"
+
+
+def safe_mkdir(dir):
+    try:
+        os.makedirs(dir)
```

Comment:
What do we need to do to ensure that this directory is always writable? Can we throw a pretty error to the user if we detect that this isn't the case? For `argo-workflows`, we get around this limitation by mounting an emptyDir volume but it seems that isn't an option in Airflow.

Reply:
Airflow communicates XCOM via a sidecar container that has `emptyDir` volume mounted to `/airflow/xcom`. There could be a condition where the side car has not started when the job container starts execution. So this condition ensures that directories are made safely. I don't think we need to raise any exception here.